### ISSUE TITLE
Issue with "::first-letter (:first-letter)": (Google search result header does not show properly for pseudo-classes)

### ISSUE THREAD

**Issue by deekeh** [mdn/content#10317]
MDN URL: https://developer.mozilla.org/en-US/docs/Web/CSS/::first-letter

#### What information was incorrect, unhelpful, or incomplete?
When we search for any pseudo-class on Google, It ignores some of the first keywords while showing MDN in the results list.

#### Specific section or headline?
![image](https://user-images.githubusercontent.com/43519784/140538564-3ce8ea77-e976-4dd1-8701-a86aa70425eb.png)
For instance, the result should show something like `::before (:before) - CSS: Cascading Style Sheets - MDN Web Docs`. But instead, it ignores "::before (:" part in the results.

#### What did you expect to see?
Some result header like `::before (:before) - CSS: Cascading Style Sheets - MDN Web Docs` should show in the results. It seems to be some issue in SEO.

#### Did you test this? If so, how?
Checked on different search Engines by searching the same text "css before".
Search engines tested on:
1. Google: failed
1. Duckduckgo: passed
1. Dogpile: passed
1. Yandex: passed (shows `before` instead of `::before`)

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/css/_doublecolon_first-letter`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/CSS/::first-letter
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/css/_doublecolon_first-letter/index.md
* Last commit: https://github.com/mdn/content/commit/26f50bdaeb1228f458d21dbdf7dc110387b20daa
* Document last modified: 2021-08-12T23:59:09.000Z

</details>

**Comment by deekeh** [comment]
added search links for preview to test

**Comment by ddbeck** [comment]
I'm guessing this is [Google rewriting page titles](https://developers.google.com/search/blog/2021/08/update-to-generating-page-titles). It appears that Google is stripping out the colons. I'd guess there's some strategy here to strip needless separators from titles, which has swallowed actual content here.

Reading their docs, I'm not sure we can do anything about this. It sounds like the biggest thing we might be able to do would be to shorten the title so that there's less incentive for Google to trim it, though I don't know if that's particularly desirable for readers.

I've tagged this as help wanted. If someone knows a bit more about SEO, I'd welcome some more concrete (and substantiated by evidence) suggestions for avoiding this.

**Comment by tannerdolby** [comment]
I had a look at [Google rewriting page titles](https://developers.google.com/search/blog/2021/08/update-to-generating-page-titles) and I saw this:

> In other instances, when encountering an extremely long title, we might select the most relevant portion rather than starting at the beginning and truncating more useful parts.

That makes me think like @ddbeck mentioned, Google is stripping out the colons from the beginning and starting at `before(..` because that marks the start of the most relevant content. Also, there are 3 common separators used in SEO when searching for keywords in titles: 
```
| (pipe)
- (hyphen)
: (colon)
```

Since the `<title>` on pages that are pseudo elements (`::after`, `::before`, `::first-letter`, etc) or pseudo classes (`:hover`, etc). It makes me really think that if a `<title>` starts with a separator, in our case a colon, it will be completely ignored. It could be that Google is skipping that part as they are recognized as separators without relevant usage. 

Resources:
- https://northward.co/articles/title-separator-symbols
- https://webmasters.stackexchange.com/questions/59797/pipes-or-colons-in-html-title-tags
- https://developers.google.com/search/blog/2021/08/update-to-generating-page-titles

**Comment by tannerdolby** [comment]
The search results for [`:hover`](https://developer.mozilla.org/en-US/docs/Web/CSS/:hover) are another example of validating that if a `<title>` starts with a separator like a colon, pipe or dash that it will be ignored. 

<img width="815" alt="Screen Shot 2021-11-18 at 11 02 43 PM" src="https://user-images.githubusercontent.com/48612525/142579653-bfb5c2cc-c305-4947-9f58-3601993f7cae.png">

The third search result tells us it keeps the colon in the title as expected because it is being used as a "complete" separator e.g. "Element: desc" instead of "half complete" which happens when the `<title>` starts with a colon separator.

**Comment by deekeh** [comment]
I think a work-around of making sure that required colons are not trimmed by Google can be using HTML encoded colon `&#58;` instead of the actual colon in the `<title>` which will show the same colon without actually using the colon. I don't know if this will work surely, but this is a way I can think of now, assuming that Google will not strip the decoded form of the `<title>`.

**Comment by tannerdolby** [comment]
Yeah, I think you're spot on @deekeh. If we use the `&#58;` html entity to represent colons where we want to display `:` but not actually use a "real" separator, I think this should do the trick. Google still might treat it as a separator and remove it, but this is definitely worth testing out.

**Comment by tannerdolby** [comment]
Spun up a small demo at https://tannerdolby.com/seo-demo, once its been indexed by Google, having a look at the SEO results in the browser should display the `<title>` tag. If anyone can manage to find get the page displayed in SEO results, do share a screenshot so we all can see :)

**Comment by deekeh** [comment]
Another way of tackling it I think of is, not starting the `<title>` with separators like colon ":" or pipe "|". Adding a descriptive text before the CSS element (`:hover` / `::before`) might do the trick.

So, instead of having the page titled as *[:hover - CSS: Cascading Style Sheets | MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/:hover)*, we can have the `<title>` something like *"CSS - Cascading Style Sheets | :hover | MDN"* which does not make it look like the title is [half-empty](https://developers.google.com/search/docs/advanced/appearance/title-link#issues); or even removing the text *Cascading Style Sheets* completely since it looks like a repetitive text in MDN page titles.
(Ref: [Google Search Central]())

**Comment by yin1999** [comment]
Maybe fixed in mdn/yari#7166
